app/code/Magento/Catalog/Model/Product/Price/PricePersistence.php

```diff
@@ -209,10 +217,10 @@ private function retrieveAffectedIds(array $skus)
         $affectedIds = [];
```

Comment:
```suggestion
        $affectedIds = [[]];
```
To prevent issue when no products ids will be returned.
It’s needed only because of added `array_merge(...$affectedIds)`